src/Components/AbilityTree/AbilityTree.java

```diff
@@ -1,83 +0,0 @@
-package Components.AbilityTree;
-
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-
-public class AbilityTree {
-
-    // A map that stores abilities and their prerequisites
-    private Map<String, String> abilityTree = new HashMap<>();
-
-    // A set that tracks unlocked abilities
-    private Set<String> unlockedAbilities = new HashSet<>();
-
-    // Adds an ability with a prerequisite to the tree
-    public void addAbility(String ability, String prerequisite) {
-        if (prerequisite == null) {
```

Comment:
I might make a prerequisite object, so you can have an default "no prerequisite" object sort of like our empty tree. No need to let the user construct this object. You can provide a kernel method that provides an empty prerequisite object like `newSequenceOfTree` in Tree.

Reply:
In general, I don't like having to deal with null or strings. Null for the null pointer exception reason and strings for their complexity (e.g., typos).